app/code/Magento/InventorySales/Model/OrderRepository.php

```diff
@@ -0,0 +1,227 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\InventorySales\Model;
+
+use Magento\Framework\Api\ExtensionAttribute\JoinProcessorInterface;
+use Magento\Framework\Api\SearchCriteria\CollectionProcessorInterface;
+use Magento\Framework\App\ObjectManager;
+use Magento\Framework\Serialize\Serializer\Json;
+use Magento\Framework\Serialize\Serializer\Json as JsonSerializer;
+use Magento\Payment\Api\Data\PaymentAdditionalInfoInterface;
+use Magento\Payment\Api\Data\PaymentAdditionalInfoInterfaceFactory;
+use Magento\Sales\Api\Data\OrderExtensionFactory;
+use Magento\Sales\Api\Data\OrderExtensionInterface;
+use Magento\Sales\Api\Data\OrderInterface;
+use Magento\Sales\Api\Data\OrderSearchResultInterfaceFactory;
+use Magento\Sales\Api\Data\ShippingAssignmentInterface;
+use Magento\Sales\Model\Order\ShippingAssignmentBuilder;
+use Magento\Sales\Model\ResourceModel\Metadata;
+use Magento\Tax\Api\OrderTaxManagementInterface;
+
+/**
+ * Replace for Order Repository in case to add Extension Attributes Join.
```

Comment:
Please add information on the following
* Deprecation notice, info on when this would happen
* Reason for introducing this class (some background on the decision)